.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDatePeriodParser.cs

```diff
@@ -1407,25 +1921,51 @@ private DateTimeResolutionResult ParseNumberWithUnit(string text, DateObject ref
                 }
             }
 
+            return ret;
+        }
+
+        private DateTimeResolutionResult ParseDuration(string text, DateObject referenceDate)
```

Comment:
Why a parseduration here, especially from text, and not reuse the existing duration parser?

Reply:
It's hard to understand the whole flow of what's calling what.

Reply:
this is analogous to the method of the same name in standard implementation and deals with cases that involve duration ("2 upcoming days", "5 previous years"). 
The internal structure was already in BaseCJKDatePeriodParser and it has been refactored here.

Reply:
Please add a code comment for it. Thanks.

Reply:
comment added